src/Files.App/ViewModels/SidebarViewModel.cs

```diff
@@ -361,189 +373,103 @@ private async Task AddElementToSection(INavigationControlItem elem, LocationItem
-								{
-									IsLibrariesHeader = true,
-									ShowHideSection = true
-								},
-								SelectsOnInvoked = false,
-								ChildItems = new BulkConcurrentObservableCollection<INavigationControlItem>()
-							};
-							var index = sectionOrder.TakeWhile(x => x != sectionType).Select(x => SideBarItems.Any(item => item.Section == x) ? 1 : 0).Sum();
-							SideBarItems.Insert(Math.Min(index, SideBarItems.Count), section);
-							section.Icon = await UIHelpers.GetIconResource(Constants.ImageRes.Libraries); // After insert
-						}
-						return section;
-					}
+		private LocationItem? GetSection(SectionType sectionType)
+		{
+			return SideBarItems.FirstOrDefault(x => x.Section == sectionType) as LocationItem;
+		}
 
-				case SectionType.Drives:
-					{
-						var section = SideBarItems.FirstOrDefault(x => x.Text == "Drives".GetLocalizedResource()) as LocationItem;
-						if (UserSettingsService.AppearanceSettingsService.ShowDrivesSection && section == null)
-						{
-							section = new LocationItem()
-							{
-								Text = "Drives".GetLocalizedResource(),
-								Section = SectionType.Drives,
-								MenuOptions = new ContextMenuOptions
-								{
-									ShowHideSection = true
-								},
-								SelectsOnInvoked = false,
-								ChildItems = new BulkConcurrentObservableCollection<INavigationControlItem>()
-							};
-							var index = sectionOrder.TakeWhile(x => x != sectionType).Select(x => SideBarItems.Any(item => item.Section == x) ? 1 : 0).Sum();
-							SideBarItems.Insert(Math.Min(index, SideBarItems.Count), section);
-							section.Icon = await UIHelpers.GetIconResource(Constants.ImageRes.ThisPC); // After insert
-						}
-						return section;
-					}
+		private async Task<LocationItem> CreateSection(SectionType sectionType)
+		{
+			LocationItem section = null;
+			BitmapImage icon = null;
+			int iconIdex = -1;
 
-				case SectionType.CloudDrives:
-					{
-						var section = SideBarItems.FirstOrDefault(x => x.Text == "SidebarCloudDrives".GetLocalizedResource()) as LocationItem;
-						if (UserSettingsService.AppearanceSettingsService.ShowCloudDrivesSection && section == null && App.CloudDrivesManager.Drives.Any())
-						{
-							section = new LocationItem()
-							{
-								Text = "SidebarCloudDrives".GetLocalizedResource(),
-								Section = SectionType.CloudDrives,
-								MenuOptions = new ContextMenuOptions
-								{
-									ShowHideSection = true
-								},
-								SelectsOnInvoked = false,
-								Icon = new BitmapImage(new Uri("ms-appx:///Assets/FluentIcons/CloudDrive.png")),
-								ChildItems = new BulkConcurrentObservableCollection<INavigationControlItem>()
-							};
-							var index = sectionOrder.TakeWhile(x => x != sectionType).Select(x => SideBarItems.Any(item => item.Section == x) ? 1 : 0).Sum();
-							SideBarItems.Insert(Math.Min(index, SideBarItems.Count), section);
-						}
-						return section;
-					}
+			if(sectionType == SectionType.Home)
```

Comment:
Can we use a switch case?

Reply:
We can, but I used the if/else pattern so I could concatenated the type of the section and the condition for it to be displayed.  Can change it if you prefer.

Reply:
I think the if statements can be put inside the cases (break early if the condition isn't met).

Reply:
Done.